test/integration/test_display_callback.py

```diff
@@ -311,3 +317,95 @@ def test_module_level_no_log(executor, playbook):
     assert len(list(executor.events))
     assert 'john-jacob-jingleheimer-schmidt' in json.dumps(list(executor.events))
     assert 'SENSITIVE' not in json.dumps(list(executor.events))
+
+
+def test_output_when_given_invalid_playbook(private_data_dir):
+    # As shown in the following issue:
+    #
+    #   https://github.com/ansible/ansible-runner/issues/29
+    #
+    # There was a lack of output by runner when a playbook that doesn't exist
+    # is provided.  This was fixed in this PR:
+    #
+    #   https://github.com/ansible/ansible-runner/pull/34
+    #
+    # But no test validated it.  This does that.
+    executor = init_runner(
+        private_data_dir=private_data_dir,
+        inventory="localhost ansible_connection=local",
+        envvars={"ANSIBLE_DEPRECATION_WARNINGS": "False"},
+        playbook=os.path.join(private_data_dir, 'fake_playbook.yml')
+    )
+
+    executor.run()
+    stdout = executor.stdout.read()
+    assert "ERROR! the playbook:" in stdout
+    assert "could not be found" in stdout
+
+
+def test_output_when_given_non_playbook_script(private_data_dir):
```

Comment:
I tried running these tests locally, and both

 - test_output_when_given_invalid_playbook
 - test_output_when_given_non_playbook_script

seem to be passing with current `devel`.

I think that makes sense because https://github.com/ansible/ansible-runner/issues/29 was closed, and PR 34 was merged.

Given what you're saying here, I think these 2 tests are fine for the purpose of adding regression coverage. However, I haven't processed the other code changes you're making here, so I can't say merge it now.

Reply:
Falls in line somewhat with what I said in [this comment](https://github.com/ansible/ansible-runner/pull/335/files#r494469400)

The first two commits of this PR:

https://github.com/ansible/ansible-runner/pull/335/commits/e3a408771ac7d7004f6ab4e1efb0e0e127013dfb
https://github.com/ansible/ansible-runner/pull/335/commits/7a155ab89411fbfa6a3d2adae0c34380f81df032

Made reference to the following PRs respectively:

- https://github.com/ansible/ansible-runner/pull/34
- https://github.com/ansible/ansible-runner/pull/256

Those PRs made the bulk of the rational of the existing block that I moved:

```python
if not self.suppress_ansible_output:
  sys.stdout.write(
    data.encode('utf-8') if PY2 else data
  )
self._handle.write(data)
self._handle.flush()
```

and I assume they were either untested via an integration test (so actually using `pexpect`), or not tested at all.  

That said, I am a bit hazy on the details at this point, but I believe the intent was that before I just moved this code, I wanted something in place that confirmed the existing fixes.  I would _**hope**_ that I had the discipline at the time to first run my tests with a original fix reverted to confirm it didn't work, then replace it to test that the test was valid... but I can't recall if I did...

Point being, these tests were simply to make sure the functionality/fixes from the aforementioned PRs still worked as intended after my fix in the third commit.